multiraft/multiraft_test.go

```diff
@@ -635,3 +635,90 @@ func TestConfigValidation(t *testing.T) {
 		t.Errorf("Unexpected error of validate: %s", err)
 	}
 }
+
+func runSubmitCommandBenchmark(numNodes, pendingCommands int, b *testing.B) {
+	stopper := stop.NewStopper()
+	defer stopper.Stop()
+	cluster := newTestCluster(nil, numNodes, stopper, b)
+	groupID := roachpb.RangeID(1)
+	cluster.createGroup(groupID, 0, numNodes)
+	if numNodes > 1 {
+		cluster.elect(0, groupID)
+	}
+
+	// Rate limiting: the producer (main) thread writes to the throttle
+	// channel before submitting a command. The consumer thread started
+	// here reads from the throttle channel as commands are committed.
+	throttle := make(chan struct{}, pendingCommands)
```

Comment:
somewhat frail, but as long as you have no reasonable suspicion that unexpected messages could show up in the committed feed (due to reelections, etc), that's fine with me.